test/functional/helpers.lua

```diff
@@ -290,6 +290,22 @@ local function expect(contents)
   return eq(dedent(contents), curbuf_contents())
 end
 
+local function rmdir(path)
+  if lfs.attributes(path, 'mode') ~= 'directory' then
+    return nil
+  end
+  for file in lfs.dir(path) do
+    if file == '.' or file == '..' then
+      goto continue
+    end
+    if not os.remove(path..'/'..file) then
```

Comment:
Yup, both would be possible.

Caller handles problems:
```lua
ret, err = func()
if not ret then
  return nil, err
end
```
Callee handles problems:
```lua
ret, err = func()
if not ret then
  print('func(): '..err)
  return nil
end
```
Or a combination of course. I prefer the callee solution, so the tests don't get cluttered with conditionals.